gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FsSpecConsumer.java

```diff
@@ -0,0 +1,127 @@
+  public static final String SPEC_PATH_KEY = "gobblin.cluster.specConsumer.path";
+  public static final String VERB_KEY = "Verb";
+
+  private final Path specDirPath;
+  private final FileSystem fs;
+  private Map<URI, Path> specToPathMap = new HashMap<>();
+
+
+  public FsSpecConsumer(Config config) {
+    this.specDirPath = new Path(config.getString(SPEC_PATH_KEY));
+    try {
+      this.fs = this.specDirPath.getFileSystem(new Configuration());
+    } catch (IOException e) {
+      throw new RuntimeException("Unable to detect spec directory file system: " + e, e);
+    }
+  }
+
+  /** List of newly changed {@link Spec}s for execution on {@link SpecExecutor}. */
+  @Override
+  public Future<? extends List<Pair<SpecExecutor.Verb, Spec>>> changedSpecs() {
+    List<Pair<SpecExecutor.Verb, Spec>> specList = new ArrayList<>();
+    FileStatus[] fileStatuses;
+    try {
+      fileStatuses = this.fs.listStatus(this.specDirPath);
+    } catch (IOException e) {
+      log.error("Error when listing files at path: {}", this.specDirPath.toString(), e);
+      return null;
+    }
+
+    for (FileStatus fileStatus : fileStatuses) {
+      DataFileReader<AvroJobSpec> dataFileReader;
+      try {
+        dataFileReader = new DataFileReader<>(new FsInput(fileStatus.getPath(), this.fs.getConf()), new SpecificDatumReader<>());
+      } catch (IOException e) {
+        log.error("Error creating DataFileReader for: {}", fileStatus.getPath().toString(), e);
+        continue;
+      }
+
+      AvroJobSpec avroJobSpec = null;
+      while (dataFileReader.hasNext()) {
+        avroJobSpec = dataFileReader.next();
+        break;
+      }
+
+      if (avroJobSpec != null) {
+        JobSpec.Builder jobSpecBuilder = new JobSpec.Builder(avroJobSpec.getUri());
+        Properties props = new Properties();
+        props.putAll(avroJobSpec.getProperties());
+        jobSpecBuilder.withJobCatalogURI(avroJobSpec.getUri()).withVersion(avroJobSpec.getVersion())
+            .withDescription(avroJobSpec.getDescription()).withConfigAsProperties(props);
+
+        try {
+          if (!avroJobSpec.getTemplateUri().isEmpty()) {
+            jobSpecBuilder.withTemplate(new URI(avroJobSpec.getTemplateUri()));
+          }
+        } catch (URISyntaxException u) {
+          log.error("Error building a job spec: ", u);
+          continue;
+        }
+
+        String verbName = avroJobSpec.getMetadata().get(VERB_KEY);
+        SpecExecutor.Verb verb = SpecExecutor.Verb.valueOf(verbName);
+
+        JobSpec jobSpec = jobSpecBuilder.build();
+        specList.add(new ImmutablePair<SpecExecutor.Verb, Spec>(verb, jobSpec));
+        this.specToPathMap.put(jobSpec.getUri(), fileStatus.getPath());
+      }
+    }
+    return new CompletedFuture<>(specList, null);
+  }
+
+
+  @Override
+  public void commit(Spec spec) throws IOException {
+    Path path = this.specToPathMap.get(spec.getUri());
+    if (path != null) {
+      this.fs.delete(path, true);
```

Comment:
Should set the recursive flag to false to be safe since only file delete is expected here.